### PR TITLE
fix timestamp differences

### PR DESCRIPTION
**Pull request by rjnrohit**
THe PR fixes dummy block addition issues for `eth_mutlicall`

### CODE REVIEW

File: src/Nethermind/Nethermind.Config/BlocksConfig.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ public class BlocksConfig : IBlocksConfig
 
         public ulong SecondsPerSlot { get; set; } = 12;
 
+        public ulong SimulateSecondsPerSlot { get; set; } = 1;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Config/BlocksConfig.cs:26]: Maybe it fits better in `JsonRcpConfig`?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Config/BlocksConfig.cs (hunk 1 above)
Comment (rjnrohit) [reply src/Nethermind/Nethermind.Config/BlocksConfig.cs:26]: Yes, we can do it if we want to make it configurable.

**Review by MarekM25 — CHANGES_REQUESTED** [review]
Have you discussed it with Geth Devs? I think 12 seconds is better. What is the reasoning for 1nd?

**Comment by LukaszRozmej** [comment]
> Have you discussed it with Geth Devs? I think 12 seconds is better. What is the reasoning for 1nd?

Yes this is actually unifying eth_simulate space. Main reasoning is to have similar cross-chain behaviour.

**Comment by rjnrohit** [comment]
> Have you discussed it with Geth Devs? I think 12 seconds is better. What is the reasoning for 1nd?

Yes, I had a call with the eth_simulate devs teams; geth has 1 second increment, you can check from here: https://github.com/s1na/go-ethereum/blob/f25fcd34e11061568c187c44dd02fb6e9c8546de/internal/ethapi/simulate.go#L48

**Comment by MarekM25** [comment]
> > Have you discussed it with Geth Devs? I think 12 seconds is better. What is the reasoning for 1nd?
> 
> Yes this is actually unifying eth_simulate space. Main reasoning is to have similar cross-chain behaviour. User can provide overrides for it if he wants.

yes, so that's why our SecondsPerSlot can be used, providing overrides sounds worse

**Comment by MarekM25** [comment]
> > Have you discussed it with Geth Devs? I think 12 seconds is better. What is the reasoning for 1nd?
> 
> Yes, I had a call with the eth_simulate devs teams; geth has 1 second increment, you can check from here: https://github.com/s1na/go-ethereum/blob/f25fcd34e11061568c187c44dd02fb6e9c8546de/internal/ethapi/simulate.go#L48

but why? I think the better approach would be to do SecondsPerSlot and this value is configurable

**Review by MarekM25 — COMMENTED** [review]
I don't think we need a seperate config value for this, either hardcode 1 and people will override it through RPC -> like geth
or just do what we are doing now and discuss with the geth team

**Comment by rjnrohit** [comment]
> I don't think we need a seperate config value for this, either hardcode 1 and people will override it through RPC -> like geth or just do what we are doing now and discuss with the geth team

for geth, it is completely hardcoded and not configurable either

**Comment by MarekM25** [comment]
> > I don't think we need a seperate config value for this, either hardcode 1 and people will override it through RPC -> like geth or just do what we are doing now and discuss with the geth team
> 
> for geth, it is completely hardcoded and not configurable either

https://github.com/s1na/go-ethereum/pull/new/simulate/timestamp-default

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs [hunk 2]
```diff
@@ -120,91 +121,67 @@ public override ResultWrapper<IReadOnlyList<SimulateBlockResult>> Execute(
                 $"Too many blocks provided, node is configured to simulate up to {_blocksLimit} while {call.BlockStateCalls?.Count} were given",
                 ErrorCodes.InvalidParams);
 
-        secondsPerSlot ??= new BlocksConfig().SecondsPerSlot;
+        ulong secondsPerSlot = new BlocksConfig().SecondsPerSlot;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:124]: you cannot rely on `new` here, as you will ignore configured value

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs [hunk 3]
```diff
@@ -120,91 +121,67 @@ public override ResultWrapper<IReadOnlyList<SimulateBlockResult>> Execute(
...
-                        StateOverrides = null,
-                        Calls = []
-                    });
-                }
-            }
-
-            call.BlockStateCalls.Sort((b1, b2) => b1.BlockOverrides!.Number!.Value.CompareTo(b2.BlockOverrides!.Number!.Value));
+            // sort element of the completeBlockStateCalls by block number
+            completeBlockStateCalls.Sort((a, b) => a.BlockOverrides.Number!.Value.CompareTo(b.BlockOverrides.Number!.Value));
+            call.BlockStateCalls = [.. completeBlockStateCalls];
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:184]: Why do we need to sort at the end? Shouldn't we just error out if it wasn't consecutive in the first place?

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs [hunk 4]
```diff
@@ -120,91 +121,67 @@ public override ResultWrapper<IReadOnlyList<SimulateBlockResult>> Execute(
...
                 ErrorCodes.InvalidParams);
 
-        secondsPerSlot ??= new BlocksConfig().SecondsPerSlot;
+        ulong secondsPerSlot = new BlocksConfig().SecondsPerSlot;
 
         if (call.BlockStateCalls is not null)
         {
-            long lastBlockNumber = -1;
-            ulong lastBlockTime = 0;
+            long lastBlockNumber = header.Number;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:128]: Can we break down this setup to some meaningful helper methods? It is hard to follow the intent.

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs (hunk 2 above)
Comment (rjnrohit) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:124]: Okay, this is what we had before, let me change it.

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs (hunk 4 above)
Comment (rjnrohit) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:128]: Sure

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs (hunk 3 above)
Comment (rjnrohit) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:184]: Yes you are correct, there is no need for sorting. Anyway we don't have to check whether the blocks are in consecutive order, as the above the block will error out/

Comment (rjnrohit) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs:184]: Yes, you are correct; we don't need to sort. We don't need to add a check either for the blocks to be consecutive as  the for loop will take care of it

**Review by LukaszRozmej — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]
